### ISSUE TITLE
[no-inferrable-types] Disable declaration of types when we can infer them

### ISSUE THREAD

**Issue by jmlopez-rod** [iOffice/tslint-config-ioffice#3]
See: https://palantir.github.io/tslint/rules/no-inferrable-types/

**Comment by rmoscowitz** [comment]
Yep, I'm cool with turning this on. I like the option too: `"no-inferrable-types": [true, "ignore-params"]` (ignore-params = "allows specifying an inferrable type annotation for function params")

**Comment by jmlopez-rod** [comment]
The `ignore-params` options allows us to do:

```ts
function someFunction(a: number, b: boolean = false) {
   // ...
}
```

Otherwise it would complain that `b` is inferred. I'm a bit conflicted in this case. Not sure if I would prefer

```ts
function anotherFunction(a: number, b = false) {
  // ...
}
```

I could argue that the type should be explicit in function declaration even for primitive types because a function declaration serves as a contract which specifies the inputs and its outputs.

**Comment by rmoscowitz** [comment]
Actually, yeah I agree, explicit types (with default values) is clearer.
